x-pack/filebeat/input/salesforce/input_manager.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package salesforce
+
+import (
+	"fmt"
+
+	"github.com/elastic/go-concert/unison"
+
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	inputcursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// inputManager wraps one stateless input manager
+// and one cursor input manager. It will create one or the other
+// based on the config that is passed.
+type InputManager struct {
+	cursor *inputcursor.InputManager
+}
+
+// NewInputManager creates a new input manager.
+func NewInputManager(log *logp.Logger, store inputcursor.StateStore) InputManager {
+	return InputManager{
+		cursor: &inputcursor.InputManager{
+			Logger:     log,
+			StateStore: store,
+			Type:       inputName,
+			Configure:  cursorConfigure,
+		},
+	}
+}
+
+// cursorConfigure configures the cursor input manager.
+func cursorConfigure(cfg *conf.C) ([]inputcursor.Source, inputcursor.Input, error) {
+	config := defaultConfig()
```

Comment:
```suggestion
	var config config
```
If it's just about creating an empty config, it'd be better to do it this way and avoid the function call overhead to do the same.

Reply:
I agree with your point. But it is kept generic for a reason like other inputs. At the moment, your point is right. But as soon as we introduce any default config (which we might have to add) then `defaultConfig` function would help. 

Although if the compiler is smart it can optimize (func call overhead) this but I'm not sure if it is doing so or not here in this case.

Reply:
> as soon as we introduce any default config (which we might have to add) then defaultConfig function would help.

In that case, couldn't we initialize the config object with whatever fields required? 
```go
defaultConfig := config{field_name: field_value} 
```

IMO this is the way it should be. I may be missing some other context, so if you still think that a function call would be better, happy to mark it as resolved.

Reply:
I think we have one field, which can be used in the default config, `version`. i will add it.
we have version validation over API less than 46, but I will add the default as 56 (current stable version).